test/index.js

```diff
@@ -0,0 +1,109 @@
+var test = require('tape')
+var encdown = require('..')
+var memdown = require('memdown')
+
+test('opens and closes the underlying db', function (t) {
+  var _db = {
+    open: function (opts, cb) {
+      t.pass('open called')
+      setImmediate(cb)
+    },
+    close: function (cb) {
+      t.pass('close called')
+      setImmediate(cb)
+    }
+  }
+  var db = encdown(_db)
+  db.open(function (err) {
+    t.error(err, 'no error')
+    db.close(function (err) {
+      t.error(err, 'no error')
+      t.end()
+    })
+  })
+})
+
+test('encodings defaults to utf8', function (t) {
+  var db = encdown(memdown())
+  t.ok(db.db, '.db should be set')
+  t.ok(db.codec, '.codec should be set')
+  t.deepEqual(db.codec.opts, {
+    keyEncoding: 'utf8',
+    valueEncoding: 'utf8'
+  }, 'correct defaults')
+  t.end()
+})
+
+test('test safe decode in get', function (t) {
+  var memdb = memdown()
+  var db = encdown(memdb, { valueEncoding: 'utf8' })
+  db.put('foo', 'this {} is [] not : json', function (err) {
+    t.error(err, 'no error')
+    var db2 = encdown(memdb, { valueEncoding: 'json' })
+    db2.get('foo', function (err, value) {
+      t.equals('EncodingError', err.name)
+      memdb.close(t.end.bind(t))
+    })
+  })
+})
+
+test('can decode from string to json', function (t) {
+  var memdb = memdown()
```

Comment:
I just realized that you can create multiple `encoding-down` and reuse the same backend. Cool! (So it's like a view?)